node/resolvers/Queries/Roles.ts

```diff
@@ -1,59 +1,52 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { currentSchema } from '../../utils'
+// import { currentSchema } from '../../utils'
 import { syncRoles } from '../Mutations/Roles'
+import { rolesVbaseId } from '../../utils'
 import { getUserByRole } from './Users'
 
-const config: any = currentSchema('b2b_roles')
+// const config: any = currentSchema('b2b_roles')
 
-export const getRole = async (_: any, params: any, ctx: Context) => {
+export const searchRoles = async (_: any, ctx: Context) => {
   const {
-    clients: { masterdata, vbase },
+    clients: { vbase },
   } = ctx
 
   try {
-    const { id } = params
-    let role: any = await vbase.getJSON('b2b_roles', id).catch(() => null)
+    const roles = (await vbase.getJSON('b2b_roles', rolesVbaseId)) as any[]
 
-    if (!role) {
-      role = await masterdata.getDocument({
-        dataEntity: config.name,
-        id,
-        fields: ['id', 'name', 'features', 'locked', 'slug'],
-      })
-    }
+    roles.sort((a, b) => {
+      return a.name > b.name ? 1 : -1
+    })
 
-    return role
+    return roles
   } catch (e) {
-    return { status: 'error', message: e }
+    if (e?.response?.status === 404) {
+      return []
+    }
+
+    throw new Error(e)
   }
 }
 
-export const searchRoles = async (_: any, params: any, ctx: Context) => {
-  const {
-    clients: { masterdata },
-  } = ctx
+export const getRole = async (_: any, params: any, ctx: Context) => {
+  try {
+    const { id } = params
 
-  const options: any = {
-    dataEntity: config.name,
-    fields: ['id', 'name', 'features', 'locked', 'slug'],
-    schema: config.version,
-    pagination: { page: 1, pageSize: 50 },
-  }
+    const role: any = (await searchRoles(null, ctx)).find(
+      (item: any) => item.id === id
+    )
 
-  if (params?.query) {
-    options.where = params?.query
+    return role
+  } catch (e) {
+    return { status: 'error', message: e }
   }
-
-  const roles = await masterdata.searchDocuments(options)
-
-  return roles
 }
 
 export const hasUsers = async (_: any, params: any, ctx: Context) => {
-  const roles: any = await searchRoles(_, { query: `slug=${params.slug}` }, ctx)
+  const role: any = await getRole(null, { id: params.slug || params.id }, ctx)
 
-  if (roles.length) {
-    const usersByRole: any = await getUserByRole(_, { id: roles[0].id }, ctx)
+  if (role) {
```

Comment:
If `getRole` encounters an error, `role` will be an object like this: `{ status: 'error', message: e }` which is still truthy. Maybe the conditional should check for `role?.id`, or alternatively `getRole` could return `null` in case of an error.